pkg/cmd/repo/create/create.go

```diff
@@ -164,11 +173,45 @@ func createRun(opts *CreateOptions) error {
 		}
 	}
 
+	// find template ID
+
+	if opts.Template != "" {
+		httpClient, err := opts.HttpClient()
+		if err != nil {
+			return err
+		}
+
+		var toView ghrepo.Interface
+		apiClient := api.NewClientFromHTTP(httpClient)
+
+		// var err errors
+		viewURL := opts.Template
+		if !strings.Contains(viewURL, "/") {
+			currentUser, err := api.CurrentLoginName(apiClient, ghinstance.Default())
+			if err != nil {
+				return err
+			}
+			viewURL = currentUser + "/" + viewURL
+		}
+		toView, err = ghrepo.FromFullName(viewURL)
+		if err != nil {
+			return fmt.Errorf("argument error: %w", err)
+		}
+
+		repo, err := api.GitHubRepo(apiClient, toView)
+		if err != nil {
+			return err
+		}
+
+		opts.Template = repo.ID
+	}
+
 	input := repoCreateInput{
 		Name:             repoToCreate.RepoName(),
 		Visibility:       visibility,
 		OwnerID:          repoToCreate.RepoOwner(),
 		TeamID:           opts.Team,
+		RepositoryID:     opts.Template,
```

Comment:
Since this type is intended to pass as input for the create repo mutation I'm not in love with using it to signal the desire to clone from a template.

I'd rather see:

- a new helper, `repoCreateFromTemplate`
- a change to `repoCreate`'s function signature to accept a potentially empty template argument

Reply:
Addressed in https://github.com/cli/cli/pull/1590/commits/f465b07f3f05392f9414da57f2ef94830291c725

@vilmibm I've refactored the changes so that `repoCreate` is a [variadic function:](https://gobyexample.com/variadic-functions)

`func repoCreate(client *http.Client, hostname string, input repoCreateInput, template ...string) (*api.Repository, error)`

As a result, I was able to remove this from the `repoCreateInput` block, and passed it as an argument instead whenever `repoCreate` was called!